### ISSUE TITLE
[BUG] Unity messages classified as unused code

### ISSUE THREAD

**Issue by jbevain** [microsoft/vscode-dotnettools#1587]
### Describe the Issue

Starting with C# DevKit 1.12.37 prerelease + C# 2.54.20 prerelease, Unity event functions/messages are categorized as unused code.

### Steps To Reproduce

1. Open or create a Unity project
2. Open or create a `MonoBehaviour`

### Expected Behavior

The `Start` and `Update` methods are categorized and visually identified as normal code.

**Comment by jbevain** [comment]
## Workaround:

As a workaround, you can stay on:

* C# DevKit v1.11.14 + C# 2.50.27

Or add a `.editorconfig` file to the root of your Unity project, containing:

```
dotnet_diagnostic. IDE0055.severity = none
dotnet_diagnostic. IDE0052.severity = none
dotnet_diagnostic. IDE0051.severity = none
dotnet_diagnostic. IDE0044.severity = none
```

Or also make the Unity messages `public`.

**Comment by arunchndr** [comment]
Keeping this open for tracking the fix rollout for https://developercommunity.visualstudio.com/t/Private-Unity-messages-incorrectly-marke/10779025 specific to C# Dev Kit/VS Code.

**Comment by dibarbet** [comment]
This should be resolved in 2.59.14 (prerelease) of the C# extension
